### PR TITLE
- add multisig support for adding or removing a signer

### PR DESCRIPTION
**Pull request by bflood-strike**
Separate multisig operations for adding or removing a signer

## Description

- added new instruction for initializing add or remove signer operation
- added new instruction for finalizing add or remove signer operation

## Motivation and Context

- Adding specific approval types for config changes

## How Has This Been Tested?

- Both success and failure unit tests have been added
- Will also be tested using backend integration tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix breaking (fix that would cause existing functionality to change)

## Checklist:

- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x ] I have added tests to cover my changes.
- [ x] All new and existing tests passed.

### CODE REVIEW

File: src/instruction.rs [hunk 1]
```diff
@@ -113,6 +114,25 @@ pub enum ProgramInstruction {
         amount: u64,
         token_mint: Pubkey,
     },
+
+    /// 0  `[writable]` The multisig operation account
+    /// 1. `[]` The program config account
+    /// 2. `[signer]` The initiator account (either the transaction assistant or an approver)
+    /// 3. `[]` The sysvar clock account
+    InitAddOrRemoveSigner {
```
Comment (aonyshchenko-strike) [inline src/instruction.rs:122]: Nit: How about naming this `UpdateSigners`? I would also change `ActionType` to `SlotUpdateType` and define it as follows:

```
pub enum SlotUpdateType {
    SET_IF_EMPTY = 0,
    CLEAR = 1,
}
```

**Comment by aonyshchenko-strike** [comment]
Looks great! 👍

**Review by aonyshchenko-strike — DISMISSED** [review]

File: src/instruction.rs (hunk 1 above)
Comment (bflood-strike) [reply src/instruction.rs:122]: yep, sure I can make that change.

Comment (bflood-strike) [reply src/instruction.rs]: made this change

**Review by aonyshchenko-strike — APPROVED** [review]
Ship it! 🚢

**Pull request merged**